abdiff/core/run_ab_transforms.py

```diff
@@ -1 +1,187 @@
 """abdiff.core.run_ab_transforms"""
+
+import logging
+import os
+import re
+import time
+from datetime import timedelta
+from pathlib import Path
+from time import perf_counter
+
+import docker
+from docker.models.containers import Container
+
+from abdiff.config import Config
+from abdiff.core.exceptions import DockerContainerRuntimeExceededTimeoutError
+from abdiff.core.utils import create_subdirectories, update_or_create_run_json
+
+CONFIG = Config()
+
+logger = logging.getLogger(__name__)
+
+
+def run_ab_transforms(
+    run_directory: str,
+    image_name_a: str,
+    image_name_b: str,
+    input_files: list[str],
+    docker_client: docker.client.DockerClient | None = None,
+    timeout: int = 180,
+) -> tuple[list[str], ...]:
+    """Run Docker containers with versioned images of Transmogrifier."""
```

Comment:
I think a very _high_ level flow added to this docstring could be helpful, something along the lines of:

```
1. Directories are created to capture the transformed files
2. For all input files, an A and B version of Transmogrifier is run via detached Docker containers, resulting in all containers running in parallel
3. Wait for all containers to complete
4. Aggregate logs from all containers
5. Return a list of all A and B transformed files
```

To the degree this is added, it may be that some comments in code then feel unneeded or redundant (but definitely keep if they feel helpful!)

Reply:
Agreed ☝️

Reply:
See [updated docstring for `run_ab_transforms`](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/6/files#diff-7a49d9dbf619fca67eb5ed5ed16b6c62440a4cc6898754cd835d2b819be74ddcR32-R71).